pyomo/contrib/fme/fourier_motzkin_elimination.py

```diff
@@ -0,0 +1,312 @@
+                for var in nonlinear_vars:
+                    if var in vars_to_eliminate:
+                        raise RuntimeError("Variable %s appears in a nonlinear "
+                                           "constraint. The Fourier-Motzkin "
+                                           "Elimination transformation can only "
+                                           "be used to eliminate variables "
+                                           "which only appear linearly." % 
+                                           var.name)
+            for var in std_repn.linear_vars:
+                if var in vars_to_eliminate:
+                    vars_that_appear.append(var)
+
+        # we actually begin the recursion here
+        while vars_that_appear:
+            # first var we will project out
+            the_var = vars_that_appear.pop()
+
+            # we are 'reorganizing' the constraints, we sort based on the sign
+            # of the coefficient of the_var: This tells us whether we have
+            # the_var <= other stuff or vice versa.
+            leq_list = []
+            geq_list = []
+            waiting_list = []
+
+            for cons in constraints:
+                leaving_var_coef = cons['map'].get(the_var)
+                if leaving_var_coef is None or leaving_var_coef == 0:
+                    waiting_list.append(cons)
+                    continue
+
+                # we know the constraint is a >= constraint, using that
+                # assumption below.
+                # NOTE: neither of the scalar multiplications below flip the
+                # constraint. So we are sure to have only geq constraints
+                # forever, which is exactly what we want.
+                if leaving_var_coef < 0:
+                    leq_list.append(
+                        self._nonneg_scalar_multiply_linear_constraint(
+                            cons, -1.0/leaving_var_coef))
+                else:
+                    geq_list.append(
+                        self._nonneg_scalar_multiply_linear_constraint(
+                            cons, 1.0/leaving_var_coef))
+
+            constraints = waiting_list
+            for leq in leq_list:
+                for geq in geq_list:
+                    constraints.append(self._add_linear_constraints(leq, geq))
+
+            # add back in the constraints that didn't have the variable we were
+            # projecting out
+            #constraints.extend(waiting_list)
+
+        return constraints
+
+    def _nonneg_scalar_multiply_linear_constraint(self, cons, scalar):
+        """Multiplies all coefficients and the RHS of a >= constraint by scalar.
+        There is no logic for flipping the equality, so this is just the 
+        special case with a nonnegative scalar, which is all we need.
+        """
+        cons['body'].linear_coefs = [scalar*coef for coef in
+                                     cons['body'].linear_coefs]
```

Comment:
Don't you need to update the quadratic and nonlinear terms as well (if you are going to allow nonlinear constraints)?